bin/PerlDDS/Run_Test.pm

```diff
@@ -611,7 +618,19 @@ sub setup_discovery {
   my $params = shift;
   my $executable = shift;
   $params = "" if !defined($params);
-  $executable = "$ENV{DDS_ROOT}/bin/DCPSInfoRepo" if !defined($executable);
+  if (!defined($executable)) {
+    $executable = "$ENV{DDS_ROOT}/bin/DCPSInfoRepo";
+    my $extension = ".exe";
+    my $executable_ext = $executable.$extension;
+    if (!(-e $executable) && !(-e $executable_ext)) {
+      if (!defined($ENV{OPENDDS_INSTALL_PREFIX})) {
+        print STDERR "ERROR: OPENDDS_INSTALL_PREFIX is not defined\n";
+        exit 1;
```

Comment:
The only thing I should see that might really have to be improved is how to handle this error case. It should probably mention that it couldn't find `$DDS_ROOT/bin/DCPSInfoRepo`. Also I'm not entirely sure if we should be doing a `exit` here as libraries really shouldn't do that. A little further down on line 644 we are saying `$self->{status} = -1;`, but I don't know what that means exactly and I'm confused because it looks like we should be returning after that, be we don't.

Reply:
I'm OK with an `exit` here since this is a library only built for running OpenDDS tests, and if we can't find OpenDDS then the overall Perl process has nothing more it can do.

I think the PR is general is just about ready to merge, but I want to make sure we have consensus on how `Run_Test.pm` should work first.

Reply:
I actually have changed it yesterday, following Fred's suggestion, but can change it back of course. Where in the test process $self->{status} = -1; get processed, and what is the outcome?